### PR TITLE
bugfix and UX enh. for tool form footer

### PR DESCRIPTION
**Pull request by martenson**
- fix citation section being always displayed
- fix tooltips not rendering on icons (related to https://github.com/galaxyproject/galaxy/pull/14857)
- touchup the order and a11y of elements

BEFORE
Galaxy

AFTER
Galaxy___martenson

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by martenson** [comment]
the jest tests pass fine locally, not sure what is with it here

**Comment by dannon** [comment]
The test that's causing the error is `FormTool`, and if you run it individually locally (`yarn jest formtool`) you'll see a relevant error (that causes the suite to fail, while the test itself actually passes its assertions).

There are at least a couple places that are using hasCitations that need to be updated and/or accounted for here, if we're dropping that prop.

File: client/src/components/Tool/ToolCard.vue [hunk 1]
```diff
@@ -121,7 +121,7 @@ const showVersions = computed(() => props.options.versions?.length > 1);
 
             <ToolFooter
                 :id="props.id"
-                :has-citations="props.options.hasCitations"
+                :has-citations="props.options.citations"
```
Comment (martenson) [inline client/src/components/Tool/ToolCard.vue:124]: @dannon this was the root of the problem, and we had tests that were confirming this is falsely correct, but it never worked

**Comment by martenson** [comment]
rebased for conflicts

**Pull request merged**